src/promptflow-core/promptflow/core/_flow.py

```diff
@@ -304,8 +306,10 @@ def __init__(
         # prompty file path
         path = Path(path)
         configs, self._template = self._parse_prompty(path)
-        configs = update_dict_recursively(configs, kwargs)
-        configs["model"] = update_dict_recursively(configs.get("model", {}), model or {})
+        with _change_working_dir(path.parent):
```

Comment:
may pass in some base_path parameter instead of change working dir, which is error prone

Reply:
fix it, pass base_path to resolve_reference